File: stacks/nodejs/react-fastify/rest/bff/src/flow/github/codeowners.ts
export type CodeownersEntry = {
  pattern: string;
  owners: string[];
};

export function parseCodeowners(content: string): CodeownersEntry[] {
  const lines = content.split(/\r?\n/);
  const entries: CodeownersEntry[] = [];

  for (const rawLine of lines) {
    const line = rawLine.trim();
    if (line === "" || line.startsWith("#")) {
      continue;
    }

    const parts = line.split(/\s+/).filter(Boolean);
    if (parts.length < 2) {
      continue;
    }

    const [pattern, ...owners] = parts;
    entries.push({ pattern, owners });
  }

  return entries;
}

File: stacks/go/net-http/rest/bff/flow/github/codeowners.go
package github

import "strings"

type CodeownersEntry struct {
	Pattern string
	Owners  []string
}

func ParseCodeowners(content string) []CodeownersEntry {
	entries := []CodeownersEntry{}
	for _, raw := range strings.Split(strings.ReplaceAll(content, "\r\n", "\n"), "\n") {
		line := strings.TrimSpace(raw)
		if line == "" || strings.HasPrefix(line, "#") {
			continue
		}
		parts := strings.Fields(line)
		if len(parts) < 2 {
			continue
		}
		entries = append(entries, CodeownersEntry{
			Pattern: parts[0],
			Owners:  parts[1:],
		})
	}
	return entries
}
